### PR TITLE
[Merged by Bors] - chore(RingTheory/Flat): remove `Algebra.Flat` and move `RingHom.Flat`

### PR DESCRIPTION
**Pull request by chrisflav**
We remove `Algebra.Flat` as it is a special case of `Module.Flat` and make `RingHom.Flat` a regular definition to align with e.g. `RingHom.Finite`. Also adds some meta properties of flat ring homomorphisms.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by alreadydone — APPROVED** [review]
LGTM!
I'll let the maintainer decide whether to add deprecated aliases for `Algebra.Flat(.self/.trans/.baseChange/.isBaseChange)`.
maintainer delegate

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**
